### PR TITLE
Add deploy id to deploy mail subject

### PR DESCRIPTION
**Pull request by rikeoka**
* Adds the deploy id to the subject of the deploy mails to enable searching through emails/jira tickets for the deploy record.

/cc @zendesk/samson

### Tasks
 - [ ] :+1: from team

### References
 - Jira link: N/A

### Risks
- Level: Low

### CODE REVIEW

**Comment by grosser** [comment]
the mailer subject ends up in jira ?

... mails are not searchable via their body or id is not in the body ?

... 👍  since it's not much clutter

File: app/mailers/deploy_mailer.rb [hunk 1]
```diff
@@ -33,7 +33,7 @@ def deploy_failed_email(deploy, emails)
   private
 
   def deploy_subject(deploy)
-    "[#{Rails.application.config.samson.email.prefix}] #{deploy.summary_for_email}"
+    "[#{Rails.application.config.samson.email.prefix}][#{deploy.id}] #{deploy.summary_for_email}"
```
Comment (grosser) [inline app/mailers/deploy_mailer.rb]: maybe `##{deploy.id}` so it stands out more ?

**Comment by rikeoka** [comment]
Yes, the subject of the mail ends up as the summary of the Jira ticket.
With how Jira displays search results it's easier to read than in body, so you don't have to click through search results.

**Comment by grosser** [comment]
:+1:

File: test/mailers/deploy_mailer_test.rb [hunk 2]
```diff
@@ -62,6 +62,7 @@ def stub_empty_changeset
 
     it 'sets a bypass subject' do
       subject.subject.must_match /BYPASS/
+      subject.subject.must_match Regexp.new deploy.id.to_s
```
Comment (grosser) [inline test/mailers/deploy_mailer_test.rb]: `must_include` ;)

Comment (rikeoka) [inline test/mailers/deploy_mailer_test.rb]: cleaned up the test:)

**Comment by grosser** [comment]
👍

**Pull request merged**